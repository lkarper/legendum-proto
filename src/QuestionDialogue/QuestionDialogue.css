.QuestionDialogue__text {
    margin: 5px;
}