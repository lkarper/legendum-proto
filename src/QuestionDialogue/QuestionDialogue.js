import React from 'react';
import PropTypes from 'prop-types';
import './QuestionDialogue.css';

const QuestionDialogue = (props) => {
    const { 
        page, 
        savedUserInput,
    } = props;
    
    if (Object.keys(page).length === 0) {
        return <p>Error: Looks like something went wrong. Check your connection and try again.</p>;
    } else if (page.dialogue_look_back) {

        /* 
            If a page contains a 'dialogue_look_back', then the app alters the dialogue based on user input; 
            the user's input is inserted into the dialogue
        */
        const dialogueArray = page.dialogue.split('|');
        const dialogueToDisplay = `${dialogueArray[0]}${savedUserInput[page.dialogue_to_look_for]}${dialogueArray[1]}`;
        return <p className='QuestionDialogue__text'>{dialogueToDisplay}</p>;
    }

    return <p className='QuestionDialogue__text'>{page.dialogue}</p>;
}

QuestionDialogue.defaultProps = {
    page: {},
    savedUserInput: {},
};

QuestionDialogue.propTypes = {
    page: PropTypes.object,
    savedUserInput: PropTypes.object,
};

export default QuestionDialogue;
