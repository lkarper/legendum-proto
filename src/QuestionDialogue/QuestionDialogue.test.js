import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import QuestionDialogue from './QuestionDialogue';

describe('QuestionDialogue component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<QuestionDialogue />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<QuestionDialogue />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
