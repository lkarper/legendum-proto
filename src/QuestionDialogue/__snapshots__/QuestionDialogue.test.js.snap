// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionDialogue component renders the UI as expected 1`] = `
<QuestionDialogue
  page={Object {}}
  savedUserInput={Object {}}
>
  <p>
    Error: Looks like something went wrong. Check your connection and try again.
  </p>
</QuestionDialogue>
`;
